cmd/cli/commands/run.go

```diff
@@ -381,6 +385,21 @@ func newRunCmd() *cobra.Command {
 				}
 			}
 
+			// Handle --detach flag: just load the model without interaction
+			if detach {
```

Comment:
Couldn't this have been placed at the beginning of the function so we don't need another `if !detach`?